### PR TITLE
[#165450135] Added badge on app shortcut

### PR DESCRIPTION
**Pull request by lmorelli986**
This PR aims to add a badge on app shortcut when there are new/unread messages

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#165450135](https://www.pivotaltracker.com/story/show/165450135): Come CIT Quando ricevo notifiche in app voglio vedere il badge nell'icona dell'applicazione


New dependencies added: <a href='https://www.npmjs.com/package/react-native-shortcutbadger'><code>react-native-shortcutbadger</code></a>.


<h2><a href="https://github.com/yangqiong/react-native-shortcutbadger#readme">react-native-shortcutbadger</a></h2>
<p>Author: yangqiong</p>
<p>Description: A React Native Moulde of ShortcutBadger that makes your Android app show the count of unread messages as a badgeon your App shortcut!</p>
<p>Homepage: <a href="https://github.com/yangqiong/react-native-shortcutbadger#readme">https://github.com/yangqiong/react-native-shortcutbadger#readme</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>almost 3 years ago</td></tr><tr><td>Last Updated</td><td>almost 3 years ago</td></tr><tr><td>License</td><td>MIT</td></tr><tr><td>Maintainers</td><td>1</td></tr><tr><td>Releases</td><td>2</td></tr><tr><td>Keywords</td><td>shortcutbadger and react-native</td></tr>
</table>

<details>
<summary><code>README</code></summary></br>

# react-native-shortcutbadger
A React Native Moulde of ShortcutBadger that makes your Android app show the count of unread messages as a badgeon your App shortcut!

The ShortcutBadger of Android Native which Version is V1.1.8.

ShortcutBadger: https://github.com/leolin310148/ShortcutBadger.

# Support launchers
Takes look at ShortcutBadger.

# Install
```npm install react-native-image-picker@latest --save```

1.In your `android/settings.gradle` file, make the following additions:
```
include ':react-native-shortcutbadger'
project(':react-native-shortcutbadger').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-shortcutbadger/android')
```

2.In your `android/app/build.gradle` file, add the `:react-native-shortcutbadgerproject` as a compile-time dependency:
```
...
dependencies{
    ...
    compile project(':react-native-shortcutbadger')
}
...
```

3.Update `MainActivity.java` to use shortcutbadger via the following changes
```
import com.yangqiong.shortcutbadger.ShortcutBadgerPackage;

public class MainActivity extends ReactActivity {
    ...
    protected List<ReactPackage> getPackages() {
        return Arrays.<ReactPackage>asList(
            new MainReactPackage(),
            new ShortcutBadgerPackage()     // new
        );
    }
    ...
}
```

# Usage
```
import { NativeModules } from 'react-native';
const ShortcutBadger = NativeModules.ShortcutBadger;
int count = 1;
ShortcutBadger.applyCount(count);
ShortcutBadger.removeCount();
```

</details>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by Undermaken** [comment]
@lmorelli986 Does this feature work on Android device too?

**Comment by lmorelli986** [comment]
@Undermaken I developed it on physical device Huawei P9 Lite and acts well. On virtual Android device not, though: there's no error but badge doesn't appear.

**Comment by Undermaken** [comment]
@lmorelli986 I guess on Android there are a lot of chaos about Home Launchers (on my device I use Google as launcher **Google Now**) and this is why on my tests I can't see any badges. However, here my tests. On iOS I tried and it works! ;)

### nexus 5x emulator - API 28
nexus5x9 0Emulator (Pie) - API 28
### nexus 5x real device - API 27
Nexus5XReal_8 1 0_resize

File: android/app/src/main/java/it/teamdigitale/app/italiaapp/MainActivity.java [hunk 1]
```diff
@@ -71,6 +76,14 @@ public void onRequestPermissionsResult(int requestCode, String[] permissions, in
       super.onRequestPermissionsResult(requestCode, permissions, grantResults);
     }
 
+    // Setup badge library showoff
+    protected List<ReactPackage> getPackages() {
```
Comment (fpersico) [inline android/app/src/main/java/it/teamdigitale/app/italiaapp/MainActivity.java]: Usually libraries get added to MainApplication. Are you sure of this change?

File: android/app/src/main/java/it/teamdigitale/app/italiaapp/MainApplication.java [hunk 2]
```diff
@@ -10,6 +10,7 @@
 import com.facebook.react.ReactApplication;
 import com.lewin.qrcode.QRScanReaderPackage;
 import com.imagepicker.ImagePickerPackage;
+import com.oblador.vectoricons.VectorIconsPackage;
```
Comment (fpersico) [inline android/app/src/main/java/it/teamdigitale/app/italiaapp/MainApplication.java]: Why this change?

File: android/app/src/main/java/it/teamdigitale/app/italiaapp/MainActivity.java (hunk 1 above)
Comment (lmorelli986) [reply android/app/src/main/java/it/teamdigitale/app/italiaapp/MainActivity.java]: I followed literally instructions here, but yes, after figuring out this project setup I moved code in the right place in 3ca589af

File: android/app/src/main/java/it/teamdigitale/app/italiaapp/MainApplication.java (hunk 2 above)
Comment (lmorelli986) [reply android/app/src/main/java/it/teamdigitale/app/italiaapp/MainApplication.java]: I cleaned up code removing unused modules. This is mostly because I tried several solutions before.

**Comment by Undermaken** [comment]
@lmorelli986 could you resolve conflict with this branch?

**Comment by lmorelli986** [comment]
@Undermaken updated to last master changes. Moreover, about some devices incompatibility, I read on https://github.com/leolin310148/ShortcutBadger/issues/198 that on pure android devices, icon badging it is not supported. I suppose this is the reason why tests you mentioned above failed.

File: android/app/build.gradle [hunk 3]
```diff
@@ -220,6 +220,7 @@ dependencies {
     compile "com.google.android.gms:play-services-gcm:8.1.0"
     compile "com.android.support:multidex:1.0.3"
     implementation fileTree(dir: "libs", include: ["*.jar"])
+    implementation "me.leolin:ShortcutBadger:1.1.22@aar"
```
Comment (Undermaken) [inline android/app/build.gradle]: Are you sure to manually include this dependency?
I guess react-native-notification already does https://github.com/zo0r/react-native-push-notification/blob/master/android/build.gradle
Here where they use the library https://github.com/zo0r/react-native-push-notification/blob/master/android/src/main/java/com/dieam/reactnativepushnotification/helpers/ApplicationBadgeHelper.java

File: ios/ItaliaApp.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -13,6 +13,8 @@
 		00C302EA1ABCBA2D00DB3ED1 /* libRCTVibration.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 00C302E41ABCB9EE00DB3ED1 /* libRCTVibration.a */; };
 		00E356F31AD99517003FC87E /* ItaliaAppTests.m in Sources */ = {isa = PBXBuildFile; fileRef = 00E356F21AD99517003FC87E /* ItaliaAppTests.m */; };
 		018DA104690D446AB4939E94 /* libRNInstabug.a in Frameworks */ = {isa = PBXBuildFile; fileRef = B59B6238889E4208B49E5C9D /* libRNInstabug.a */; };
+		0A2B90C3DF6444AEB67B60BD /* Octicons.ttf in Resources */ = {isa = PBXBuildFile; fileRef = 40FC08429CCB45218C278983 /* Octicons.ttf */; };
```
Comment (Undermaken) [inline ios/ItaliaApp.xcodeproj/project.pbxproj]: Since no library or modules is added, are these new resources needed by this PR?

**Comment by lmorelli986** [comment]
Feature is no longer needed. Closed.
